msbrainpy/atlas/query.py

```diff
@@ -142,3 +146,165 @@ def generateQueryRows(model, criteria, options='[only$eq''genes.entrez_id,data_s
+    url_1 = '[data_set_id$eq{}]'.format(data_set_id)
+    url = url_0 + url_1
+    response = requests.get(url)
+    image_json = response.json()
+    if verbose:
+        print('{} images matching the criteria were found'.format(image_json['num_rows']))
+    return image_json
+
+
+def download_section_images(section_image_json, out_dir_name, check_img_exists=False, return_img_n=None, **kwargs):
+    image_ids = [row['id'] for row in section_image_json['msg']]
+    if not os.path.exists(out_dir_name):
+        os.mkdir(out_dir_name)
+    for i in range(len(image_ids)):
+        img = image_ids[i]
+        filename = 'image_id-' + str(img) + '.jpeg'
+        filepath = os.path.join(out_dir_name, filename)
+        if not check_img_exists or not os.path.exists(filepath):
+            if i == return_img_n:
+                img_n = download_image(img, filepath, return_image=True)
+            else:
+                download_image(img, filepath)
+    if return_img_n is not None:
+        return img_n
+
+
+def download_image(image_id, filename, return_image=False):
+    url = 'http://api.brain-map.org/api/v2/image_download/{}'.format(image_id)
+    response = requests.get(url)
+    with open(filename, 'wb') as file:
+        file.write(response.content)
+    if return_image:
+        img = io.imread(filename)
+        return img
+
+
+# ------------------------------------------------ Get Gridded Data ----------------------------------------------------
+
+def get_gridded_data(entrez_id, directory_path=None, age_id=None, product_id=1, verbose=False):
+    download_from_section_data_set(get_section_grid, entrez_id, directory_path=directory_path, age_id=age_id, 
+                                   product_id=product_id, verbose=verbose)
+
+
+def get_section_grid(section_id, out_dir):
+    # this should obtain energy, intensity, and density but only obtains energy. Not sure why.
+    url = "http://api.brain-map.org/grid_data/download/{}include=energy,density,intensity".format(section_id)
+    response = requests.get(url)
+    filename = out_dir + '.zip'
+    with open(filename, 'wb') as file:
+        file.write(response.content)
+    with ZipFile(filename, 'r') as zipObj:
+        zipObj.extractall(out_dir)
+
+
+# ----------------------------------------------- Find Data Sets -------------------------------------------------------
+def download_from_section_data_set(func, entrez_id, directory_path=None, age_id=15, product_id=1, verbose=False):
+    section_data_set = section_data_set_query(entrez_id, plane_of_section=None, product_id=product_id,
+                                              verbose=verbose)
+    dirname = 'entrez_id_' + str(entrez_id) + '_' + section_data_set['msg'][0]['genes'][0]['acronym']
+    if directory_path is not None:
+        dirname = os.path.join(directory_path, dirname)
+
+    if not os.path.exists(dirname):
+        os.mkdir(dirname)

```

Comment:
```suggestion
    os.makedirs(dirname, exist_ok=True)
```

https://docs.python.org/3/library/os.html#os.makedirs